src/ui/MainWindow.cpp

```diff
@@ -699,6 +699,13 @@ void MainWindow::openEditorTab(const QString& absolutePath)
 {
     if (activateEditorTabByFilePath(absolutePath)) return;
 
+    if(Q_UNLIKELY(!CEGUIProjectManager::Instance().makeOpenGLContextCurrent())) // needed if a Window has "AutoRenderingSurface" set, as that will call Window::allocateRenderingWindow()
```

Comment:
This sould not be there. You must set context _immediately before_ using it, not introducing unnecessary dependencies in the rest of the application. Vast majority of editors don't require CEGUI OpenGL context.

Reply:
You don't have a crash with the legacy opengl renderer if you close and then open a project? I get it in OpenGL3FBOTextureTarget::initialiseRenderTexture() - glGenFramebuffers() because there is no active context

Reply:
No crash even if closing project with active layout rendering on screen. Anyway MainWindow::openEditorTab is not a place to set OpenGL context because there are lots of editor tabs which don't require OpenGL context and CEGUI instance at all.
I think it is better to add OpenGL3 renderer support and then fix problems that will arise. But to be honest, OpenGL3 is not in my priority now. I focus on making an usable editor, no matter how exactly (and in particulat how fast) it renders CEGUI while it doesn't affect user experience.
When editor is done we can put more effort into rendering, but it really adds to editor less than it takes from developers.